packages/gatsby-plugin-fastify/src/plugins/server-routes.ts

```diff
@@ -107,6 +109,13 @@ export const handleServerRoutes: FastifyPluginAsync<{
 
             const results = await renderHTML({ data: pageQueryData });
 
+            if (page.mode === "DSG") {
+              mapValues(NEVER_CACHE_HEADER, (value, key) => {
```

Comment:
Can this not use the `appendRouteHeaders` function?

Reply:
I did remove it initially and just had the `appendRouteHeaders` call below it but then realized `NEVER_CACHE_HEADERS` was not a part of `appendRouteHeaders` but DSG pages needed it, so I added it back in. I think this will change as I refactor and clean it up.